src/content/en/updates/2017/12/modulepreload.md

```diff
@@ -0,0 +1,138 @@
+helping you reduce the number of bytes you need to ship to your users with a new release.
+The finer granularity of the code also helps with the loading story, by letting you
+prioritise the critical code in your application.
+
+That said, dependency trees can get deep very quickly, and there wasn't really a
+good way of declaratively preloading modules until now.
+Chrome 64 ships with `<link rel="modulepreload">` enabled by default, a module-specific
+version of `<link rel="preload">` that solves a number of the latter's problems.
+
+Warning: It's still very much early days for modules in the browser, so while we
+encourage experimentation, we definitely do not recommend using this technology in
+production just yet!
+
+## Wait, what's `<link rel="preload">`?
+
+Chrome added support for `<link rel="preload">` back in version 50, as a way of
+declaratively requesting resources ahead of time, before the browser needs them.
+
+```html
+<head>
+  <link rel="preload" as="style" href="critical-styles.css">
+  <link rel="preload" as="font" crossorigin="crossorigin" type="font/woff2" href="myfont.woff2">
+</head>
+```
+
+This works particularly well with resources such as fonts, which are often hidden
+inside CSS files, sometimes several levels deep. In that situation, the browser
+would have to wait for multiple roundtrips before finding out that it needs to
+fetch a large font file, when it could have used that time to start the download
+and take advantage of the full connection bandwidth.
+
+`<link rel="preload">` and its HTTP header equivalent provide a simple, declarative
+way of letting the browser know straight away about critical files that will be needed
+as part of the current navigation. When the browser sees the preload, it starts a high
+priority download for the resource, so that by the time it's actually needed it's either
+already fetched or partly there.
+
+## OK, so why doesn't `<link rel="preload">` work for modules?
+
+This is where things get tricky. There are several credentials modes for
+resources, and in order to get a cache hit they must match, otherwise you end up
+fetching the resource twice. Needless to say, double-fetching is bad, because it
+wastes the user's bandwidth and makes them wait longer, for no good reason.
+
+For `<script>` and `<link>` tags, you can set the credentials mode with the `crossorigin`
+attribute. However, it turns out while that a `<script type="module">` with no
+`crossorigin` attribute means that the credentials mode is `omit`, which doesn't exist
+for `<link rel="preload">`. This means that you would have to
+change the `crossorigin` attribute in both your `<script>` and `<link>` to one
+of the other values, and you might not have an easy way of doing so if what you're
+trying to preload is a dependency of other modules.
+
+Furthermore, fetching the file is only the first step in actually running the code;
+it first has to be parsed and compiled by Chrome's JavaScript engine - V8. Ideally,
```

Comment:
code. First, the browser has to parse and compile....

Reply:
Done.